cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1227,6 +1232,45 @@ private boolean hasRebalanceAnnotation(KafkaRebalance kafkaRebalance) {
                 kafkaRebalance.getMetadata().getAnnotations().containsKey(ANNO_STRIMZI_IO_REBALANCE);
     }
 
+    private boolean shouldRenewRebalance(KafkaRebalance kafkaRebalance, KafkaRebalanceAnnotation rebalanceAnnotation) {
+        return  rebalanceAnnotation == KafkaRebalanceAnnotation.refresh
+                || kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> CruiseControlIssues.clusterLabelMissing.getReason().equals(condition.getReason()))
```

Comment:
This feels very verbose. You could move this checking logic into the enum so that you have something like:
```java
CruiseControlIssues.clusterLabelMissing.check(kafkaRebalance.getStatus().getConditions())
```
But the issue is you are streaming all the conditions 3 times (and will add more, if you add more condition in future). It would be more efficient to do one pass over the conditions (in a good old fashioned for loop or a stream if you want to be fancy) and check if each condition's reason matches one of the list of CruiseControlIssues instances?